services/construction_service.go

```diff
@@ -365,6 +369,54 @@ func (s *ConstructionAPIService) ConstructionMetadata(
 		return nil, wrapErr(ErrGeth, err)
 	}
 
+	// Build eth transaction for L1 fee calculation
+	var toAddress common.Address
+	var ethTx *ethTypes.Transaction
+	if to != "" {
+		toAddress = common.HexToAddress(to)
+	}
+	if isEIP1559 := gasTipCap != nil && gasFeeCap != nil; isEIP1559 {
+		ethTx = ethTypes.NewTx(&ethTypes.DynamicFeeTx{
```

Comment:
The [services.AsEthTransaction](https://github.com/Inphi/optimism-rosetta/blob/master/services/utils.go#L52) utility function can construct an appropriate `types.Transaction` based on the provided fee parameters. You can use that instead to dedup code.

Reply:
done